lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -6287,6 +6287,28 @@ LLVMToSPIRVBase::transBuiltinToInstWithoutDecoration(Op OC, CallInst *CI,
     return BM->addCompositeConstructInst(transType(CI->getType()), Operands,
                                          BB);
   }
+  case OpGroupNonUniformShuffleDown: {
+    Function *F = CI->getCalledFunction();
+    if (F->arg_size() && F->getArg(0)->hasStructRetAttr()) {
+      StructType *St = cast<StructType>(F->getParamStructRetType(0));
```

Comment:
In general it's SYCL details, that `half` (and also `bfloat16`) are hidden within classes. There should be checks, like `isSYCLHalfType` around in the translator, should we add them here to error out, if just a random structure is passed to the instruction?